distro.go

```diff
@@ -172,13 +193,38 @@ func (d Distro) GetConfiguration() (c Configuration, e error) {
 	return conf, nil
 }
 
-// String deserializes a distro and its configuration as a yaml string.
+// String deserializes a distro its GUID and its configuration as a yaml string.
+// If there is an error, it is printed as part of the yaml.
 func (d Distro) String() string {
+	return fmt.Sprintf("name: %s\n%s\n%s", d.Name(), d.guidToString(), d.configToString())
+}
+
+// guidToString shows the GUID as a yaml string.
+// It exists to simplify the implementation of (Distro).String
+// If it errors out, the message is returned as the value in the yaml.
+func (d Distro) guidToString() string {
+	registered, err := d.IsRegistered()
```

Comment:
It looks a little bit weird to return error messages here, but I see what you want to do for Stringifying unregistered distro or when we can’t get the guid. Let’s keep it that way for now, I raise my concern just for the "told you so" if we need to change it later :p

Reply:
Well the `String` method must return a string only, so I figured the best way to report the errors is to put them there. That's also part of why I argued for only printing the name: It's the only field which we are certain we can access.

Reply:
don’t get me wrong, I understood the rationale even if I didn’t tell it explicitely :)